sdk/identity/azure-identity/samples/CMakeLists.txt

```diff
@@ -8,26 +8,46 @@ set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
 add_executable(chained_token_credential_sample chained_token_credential.cpp)
-target_link_libraries(chained_token_credential_sample PRIVATE azure-identity service)
+target_link_libraries(chained_token_credential_sample PRIVATE azure-identity)
 target_include_directories(chained_token_credential_sample PRIVATE .)
+target_include_directories(
+    chained_token_credential_sample PRIVATE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../../../samples/helpers/service/inc>
+)
 create_per_service_target_build_for_sample(identity chained_token_credential_sample)
 
 add_executable(client_certificate_credential_sample client_certificate_credential.cpp)
-target_link_libraries(client_certificate_credential_sample PRIVATE azure-identity service get-env-helper)
+target_link_libraries(client_certificate_credential_sample PRIVATE azure-identity get-env-helper)
 target_include_directories(client_certificate_credential_sample PRIVATE .)
+target_include_directories(
+    client_certificate_credential_sample PRIVATE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../../../samples/helpers/service/inc>
+)
 create_per_service_target_build_for_sample(identity client_certificate_credential_sample)
 
 add_executable(client_secret_credential_sample client_secret_credential.cpp)
-target_link_libraries(client_secret_credential_sample PRIVATE azure-identity service get-env-helper)
+target_link_libraries(client_secret_credential_sample PRIVATE azure-identity get-env-helper)
 target_include_directories(client_secret_credential_sample PRIVATE .)
+target_include_directories(
+    client_secret_credential_sample PRIVATE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../../../samples/helpers/service/inc>
```

Comment:
We decided not to do that for CMake files long time ago, as per Charlie's advice (since we were doing the C SDK)

I remember because I used to set all re-usable strings from a CMake file in the top and then I would just use the variables everywhere.

So, yeah, I am not against it, but I have been following that advise of keeping the duplication. 
The only real benefit of the variables was to make it easier to update in the future, but using any editor from this days can select and replace all occurrences at the same time (and I couldn't argue with that )

Reply:
@barcharcraz can you please share some guidance and rationale for when and why we should avoid creating reusable string variables within CMake files?